### PR TITLE
Adds option to use self-signed certificates with environment variable REQWEST_USE_INSECURE_CERTS

### PR DESCRIPTION
**Pull request by aserebryakov**
If you use self-signed certificate, the following error appears when you use `AsyncTelegramApi`:

```
Failed to get updates: HttpError(HttpError { code: 500, message: "error sending request for url (https://api.telegram.org/bot5322455292:AAF0lBHoR4x1HdwrG8eZ8iJwrp4DZhvWPdU/getUpdates): error trying to connect: invalid peer certificate contents: invalid peer certificate: UnknownIssuer" })
```

So I added an option to use `REQWEST_USE_INSECURE_CERTS` environment variable to be set (value doesn't matter) to allow using such certificates.

### CODE REVIEW

**Comment by ayrat555** [comment]
@aserebryakov hello. Your changes look ok.

what do you think about passing a full client ? for example

```rust
   pub fn new_with_client(api_client: Client, api_url: String) -> Self {
        Self { api_url, client }
    }
```

and `reqwest` can be exported in the `lib.rs` so its client can be instantiated

```rust
#[doc(hidden)]
#[cfg(any(feature = "async-http-client"))]
pub use reqwest;
```

**Comment by aserebryakov** [comment]
I am quite new for Rust, but I can try to do that. Also I see that I need to fix failed checks.

I guess `unwrap()` isn't a good idea to be used in `new()` call.

**Comment by aserebryakov** [comment]
@arkada38, Please review changes. I finally implemented it.

**Comment by arkada38** [comment]
@aserebryakov, I can't merge pull requests, I don't have a [write access](https://docs.github.com/articles/what-are-the-different-access-permissions).

File: src/lib.rs [hunk 1]
```diff
@@ -28,6 +28,10 @@ pub use api::*;
 #[cfg(any(feature = "telegram-trait", feature = "async-telegram-trait"))]
 pub use api_traits::*;
 
+#[doc(hidden)]
+#[cfg(any(feature = "async-http-client"))]
```
Comment (EdJoPaTo) [inline src/lib.rs:32]: The any can be removed here as its only a single requirement here.

```suggestion
#[cfg(feature = "async-http-client")]
```

**Review by EdJoPaTo — COMMENTED** [review]
Looks good! Just some minor notes from me.

As the changes were force pushed its hard to follow the comments and understand the discussion. To me its more helpful to add commits for the changes. The full PR can then later be added as one commit so changes made wont be visible then anymore. Just as a minor note from me :)

File: src/api/async_telegram_api_impl.rs [hunk 2]
```diff
@@ -21,6 +21,10 @@ impl AsyncApi {
         Self { api_url, client }
     }
 
+    pub fn new_with_client(client: reqwest::Client, api_url: String) -> Self {
```
[the remark this answers is not in the document]
Comment (aserebryakov) [reply src/api/async_telegram_api_impl.rs:24]: Just for better understanding:

1. What is the point of adding `const` to return value? In `C++` it is usually bad idea since it prevents some of optimizations that can be applied so the guideline is use `const` variables to store the returned value.
2. Should it be done for other constructors as well to keep the same look for all?

[the remark this answers is not in the document]
Comment (aserebryakov) [reply src/api/async_telegram_api_impl.rs:24]: So if we're talking about `const` functions it's like `constexpr` functions in `C++` that allows in case of arguments that are known at compile-time calculate the result during the compilation. Interesting, I didn't know that.

**Comment by ayrat555** [comment]
@aserebryakov @EdJoPaTo Thank you!

the changes were released in 0.15.1
